src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,65 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  query: string;
+  filterType: string;
+  setQuery: React.Dispatch<React.SetStateAction<string>>;
+  setFilterType: React.Dispatch<React.SetStateAction<string>>;
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter = ({
+  query,
+  filterType,
+  setQuery,
+  setFilterType,
+}: Props) => {
+  const clearQuery = () => setQuery('');
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={filterType}
+            onChange={(event => setFilterType(event.target.value))}
+          >
+            <option value="all">All</option>
+            <option value="active">Active</option>
+            <option value="completed">Completed</option>
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={query}
+          onChange={event => setQuery(event.target.value)}
         />
-      </span>
-    </p>
-  </form>
-);
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        <span
+          className="icon is-right"
+          style={{ pointerEvents: 'all' }}
+        >
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label   */}
```

Comment:
use aria-label here 
https://stackoverflow.com/questions/58367013/a-control-must-be-associated-with-a-text-label